domain/operation/types.go

```diff
@@ -3,14 +3,50 @@
 
 package operation
 
-import "github.com/juju/juju/internal/uuid"
+import (
+	"time"
 
-// Action represents a domain action.
-type Action struct {
-	// UUID is the action unique identifier.
+	"github.com/juju/juju/internal/uuid"
+)
+
+// TaskLogMessage represents a logged message on an task.
+type TaskLogMessage struct {
+	// Timestamp is when the message was logged.
+	Timestamp time.Time
+	// Message is the content of the logged message.
+	Message string
+}
```

Comment:
I have a PR up which includes a commit to rename and use related struct in `core/operation`. Should use that one as encoding and decoding will be necessary in both the facade and cmd code. #20600